File: Convert 1D Array Into 2D Array/solution.js
var construct2DArray = function(original, m, n) {
    if (m*n !== original.length) return [];
    let result = [];
    let counter = 0;
    for (let i = 0; i < m; i++) {
        let temp = [];
        for (let j = 0; j < n; j++) {
            temp.push(original[counter]);
            counter++;
            if (j === n-1) result.push(temp);
        }
    }
    return result;
};

File: Convert 1D Array Into 2D Array/solution.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if m * n != len(original):
            return []
        result = []
        counter = 0
        for _ in range(m):
            temp = []
            for i in range(n):
                temp.append(original[counter])
                counter += 1
                if i == n-1:
                    result.append(temp)
        return result
